python/python/lancedb/query.py

```diff
@@ -895,40 +926,69 @@ class LanceHybridQueryBuilder(LanceQueryBuilder):
     def __init__(
         self,
         table: "Table",
-        query: str,
-        vector_column: str,
+        query: str = None,
+        vector_column: str = None,
         fts_columns: Union[str, List[str]] = [],
     ):
         super().__init__(table)
-        vector_query, fts_query = self._validate_query(query)
-        self._fts_query = LanceFtsQueryBuilder(
-            table, fts_query, fts_columns=fts_columns
-        )
-        vector_query = self._query_to_vector(table, vector_query, vector_column)
-        self._vector_query = LanceVectorQueryBuilder(table, vector_query, vector_column)
+        self._query = query
+        self._vector_column = vector_column
+        self._fts_columns = fts_columns
         self._norm = "score"
         self._reranker = LinearCombinationReranker(weight=0.7, fill=1.0)
+        self._nprobes = None
+        self._refine_factor = None
 
-    def _validate_query(self, query):
-        # Temp hack to support vectorized queries for hybrid search
-        if isinstance(query, str):
-            return query, query
-        elif isinstance(query, tuple):
-            if len(query) != 2:
-                raise ValueError(
-                    "The query must be a tuple of (vector_query, fts_query)."
-                )
-            if not isinstance(query[0], (list, np.ndarray, pa.Array, pa.ChunkedArray)):
-                raise ValueError(f"The vector query must be one of {VEC}.")
-            if not isinstance(query[1], str):
-                raise ValueError("The fts query must be a string.")
-            return query[0], query[1]
-        else:
+    def _validate_query(self, query, vector=None, text=None):
+        if query is not None and vector is not None:
             raise ValueError(
-                "The query must be either a string or a tuple of (vector, string)."
+                "You can either provide query or its vector representation"
+                "but not both."
             )
 
+        vector_query = query or vector
+        if not isinstance(vector_query, (str, list, np.ndarray)):
+            raise ValueError("Vector query must be either a string or a vector")
+
+        text_query = query or text
```

Comment:
I think here we need to exchange the order of parameters, consider these cases:
`table.search(some_string).text(fts_query)`
then it's supposed to `vector_query=some_string` and `text_query=fts_query`
and for `table.search(some_string).vector(vec_query)`
then it's supposed to `vector_query=vec_query` and `text_query=some_string`

Reply:
Right..makes sense